sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/AbstractScheduler.java

```diff
@@ -127,10 +127,13 @@ public ClientStatusResponse getClientStatus() {
                     inProgressSteps.stream().map(step -> step.getMessage()).collect(Collectors.toList()));
         }
         activeWorkSet.addAll(inProgressSteps);
-        if (workSetTracker == null) {
-            logger.warn("workset tracker is uninitialized, scheduler will hit a null reference and exit.");
+
+        try {
+            workSetTracker.updateWorkSet(activeWorkSet);
+        } catch (NullPointerException e) {
+            logger.warn("workset tracker is uninitialized, scheduler has his a null reference and is exiting.");
```

Comment:
Won't this catch will prevent the scheduler from exiting?

Reply:
yes, need to add an exit